### DISCUSSION TITLE
Automatically convert materials when you click the green convert number.

### DISCUSSION THREAD

**Discussion by machasins** [xanthics/genshin_grind_planner#1] [Ideas]
I use this website a lot for Genshin, but I think it could use some automation for converting materials.

I usually input all changes to my materials about once a week and then use the website to know how many materials to convert into higher tier materials. With the current system, after I converting I have to reinput all the materials I have or keep track as I go. 

I think just clicking the green convert number should automatically add the converted materials to the higher tier material's count and update the lower tier's number. _Green convert number refers to the number of materials you can convert up, next to the material number._

Accidentally auto converting materials in this system could be remedied by having an undo button where the green convert number was.

Sucrose, Albedo, Xingqiu, etc.'s talents could be introduced as a corner popup asking how many extra materials the player got. It could also be an input where the green convert number was or a more in-your-face centered popup.

Accidental clicks and talents could also just be ignored, requiring the user to manually input the gained materials.

(Also, I'm colorblind and the red and orange used for the convert numbers are almost indistinguishable. Please make the orange lighter or change it to a dim yellow. Thanks!)

Keep up the great work! Love the site :)

**Comment by xanthics** [comment]
I can change the highlight colors easy enough.  I'll push something out later today, let me know if it helps.

I'll look in to how much work there is to add an upgrade dialog.  As when I upgrade, it's often not the full amount and instead is only enough to finish a specific action.  There is also the issue of being able to convert items between types now, which would need to be added for that dialog to feel complete.

**Reply by machasins** [reply]
The new orange is much better, thank you :)
